packages/squiggle-lang/src/rescript/ReducerInterface/ReducerInterface_GenericDistribution.res

```diff
@@ -66,6 +66,38 @@ module Helpers = {
       dist1,
     )->runGenericOperation
   }
+  let parseNumber = (args: expressionValue) : Belt.Result.t<float, string> => 
+    switch args {
+    | EvNumber(x) => Ok(x)
+    | _ => Error("Not a number")
+    }
+
+  let parseNumberArray = (ags: array<expressionValue>) : Belt.Result.t<array<float>, string> => E.A.fmap(parseNumber, ags) |> E.A.R.firstErrorOrOpen
+
+  let parseDist = (args: expressionValue): Belt.Result.t<GenericDist_Types.genericDist, string> => 
+    switch args {
+    | EvDistribution(x) => Ok(x)
+    | EvNumber(x) => Ok(GenericDist.fromFloat(x))
+    | _ => Error("Not a distribution")
+    }
+
+  let parseDistributionArray = (ags: array<expressionValue>) : Belt.Result.t<array<GenericDist_Types.genericDist>, string> => E.A.fmap(parseDist, ags) |> E.A.R.firstErrorOrOpen
+
+  let mixture = (args : array<expressionValue>): DistributionOperation.outputType => {
+    let givenWeights = E.A.last(args)
+    let calculatedWeights = 
+        switch givenWeights {
+        | Some(EvArray(b)) => parseNumberArray(b)
+        | None =>
+            Ok(Belt.Array.make(E.A.length(args), 1.0 /. Belt.Int.toFloat(E.A.length(args))))
```

Comment:
I don't get when this would be ``None``. I'd assume the one case is when there are 0 arguments.

Reply:
Oh goodness, that was just a mistake

Reply:
I should have included more tests

Reply:
Ok, this is fixed in the latest commit